elastic/elastic-agent
# Add ability to split integration tests into different groups

## What does this PR do?



Adds a required field `Group` to `define.Requirements`. Adds ability to select a specific group of tests with `TEST_GROUPS` environment variable. Groups batches by `Group` name splitting the testing load across multiple instances.

## Why is it important?



Allows tests to be grouped to different instances. Allowing more tests to run concurrently to speed up the time it takes for integration tests to run. Allows the selection of running only a group of tests.

## Checklist




## How to test this PR locally



Run `mage integration:test` see that multiple instances are spawned instead of only one per platform OS/architecture.

Select a specific group with `TEST_GROUPS="default" mage integration:test`.
